build/package/.atlascli.goreleaser.yml

```diff
@@ -115,8 +111,23 @@ changelog:
     - '^test:'
     - '^task:'
     - '^build(deps):'
+publishers:
```

Comment:
what's the main reason we don't use https://goreleaser.com/customization/sign/ ? is it the podman login?

Reply:
yes, that is one of the reason. I tried to use it but I did not manage to use the login command. We also need to use the gpg provided by dev prod at `artifactory.corp.mongodb.com/release-tools-container-registry-local/garasign-gpg `

Reply:
the docs mention anything that produces a binary should be good to use, for the podman login you could try and run it before running go releaser

Reply:
DevProd team confirmed that we need to use the provided GPG binary because GPC doesn't support PGCS11

` It is important that you use the patched GPG binary though, because GPG doesn't support PKCS11 without it`

Reply:
> the docs mention anything that produces a binary should be good to use, for the podman login you could try and run it before running go releaser

I am not sure how to test it since I cannot use evergreen on the fork and, even if run a personal patch against master via evergreen, the custom publisher won't run as part of the snapshot

Reply:
@andreaangiolillo given you are changing this structure I wonder if we should consider moving the windows stuff into publishers as well

Reply:
I don't have a strong opinion. The main reason why I used publishers is that `nfpms` does not have a `hooks` field

Reply:
`podman login` is not something you need to do every-time you run podman, you should be able to log in once outside of the goreleaser flow
Then I would expect to be able to pass `podman run` as the cmd for the sign command

Reply:
@andreaangiolillo another point here is you don't have hooks on `nfpms` but you do under `build`, today we sign mac this way, we call the hook twice (for each binary) and we only sign the two at once after both are available.

in theory, you could sign each file on each run of the hook, this way you can do it even for snapshots.

Reply:
my understanding is that [nfpms](https://goreleaser.com/customization/nfpm/?h=nfpms) is where goreleaser generates the deb, .rpm, .apk, and Archlinux packages. Am I missing something here? 🤔

Reply:
@andreaangiolillo you are correct, I guess the binary is shared across packages as oppose to be generated multiple times

Reply:
> Then I would expect to be able to pass podman run as the cmd for the sign command

I tried to run `podman` as part of the command but got stuck when mounting the current directory into the container. 

`the working directory ' $(pwd)' is invalid, it needs to be an absolute path.`  as it is not able to run ' $(pwd)' as part of the flag field.

Happy to create a follow-up ticket if you have a strong opinion about trying to use the signing field of goreleaser.